airbyte-integrations/connectors/destination-oracle/src/test-integration/java/io/airbyte/integrations/destination/oracle/OracleIntegrationTest.java

```diff
@@ -0,0 +1,194 @@
+    result.add(identifier);
+    result.add(resolved);
+    if (!resolved.startsWith("\"")) {
+      result.add(resolved.toLowerCase());
+      result.add(resolved.toUpperCase());
+    }
+    return result;
+  }
+
+  private List<JsonNode> retrieveRecordsFromTable(String tableName, String schemaName) throws SQLException {
+    List<org.jooq.Record> result = Databases.createOracleDatabase(db.getUsername(), db.getPassword(), db.getJdbcUrl())
+        .query(ctx -> ctx
+            .fetch(String.format("SELECT * FROM %s.%s ORDER BY %s ASC", schemaName, tableName, JavaBaseConstants.COLUMN_NAME_EMITTED_AT.substring(1)))
+            .stream()
+            .collect(Collectors.toList()));
+    return result
+        .stream()
+        .map(r -> r.formatJSON(JSON_FORMAT))
+        .map(Jsons::deserialize)
+        .collect(Collectors.toList());
+  }
+
+  private static Database getDatabase(JsonNode config) {
+    // todo (cgardens) - rework this abstraction so that we do not have to pass a null into the
+    // constru
+    // ctor. at least explicitly handle it, even if the impl doesn't change.
+    return Databases.createDatabase(
+        config.get("username").asText(),
+        config.get("password").asText(),
+        String.format("jdbc:oracle:thin:@//%s:%s/%s",
+            config.get("host").asText(),
+            config.get("port").asText(),
+            config.get("sid").asText()),
+        "oracle.jdbc.driver.OracleDriver",
+        null);
+  }
+
+  // how to interact with the mssql test container manually.
+  // 1. exec into mssql container (not the test container container)
+  // 2. /opt/mssql-tools/bin/sqlcmd -S localhost -U SA -P "A_Str0ng_Required_Password"
+  @Override
+  protected void setup(TestDestinationEnv testEnv) throws SQLException {
+    configWithoutDbName = getConfig(db);
+    final String dbName = "db_" + RandomStringUtils.randomAlphabetic(10).toLowerCase();
+
+    final Database database = getDatabase(configWithoutDbName);
+    database.query(ctx -> {
+      ctx.execute("alter database default tablespace users");
+      ctx.execute(
+          "declare c int; begin select count(*) into c from user_tables where upper(table_name) = upper('id_and_name'); if c = 1 then execute immediate 'drop table id_and_name'; end if; end;");
+      ctx.fetch("CREATE TABLE id_and_name(id INTEGER NOT NULL, name VARCHAR(200), born TIMESTAMP WITH TIME ZONE)");
+      ctx.fetch(
+          "INSERT ALL INTO id_and_name (id, name, born) VALUES (1,'picard', TIMESTAMP '2124-03-04 01:01:01') INTO id_and_name (id, name, born) VALUES  (2, 'crusher', TIMESTAMP '2124-03-04 01:01:01') INTO id_and_name (id, name, born) VALUES (3, 'vash', TIMESTAMP '2124-03-04 01:01:01') SELECT 1 FROM DUAL");
+      return null;
+    });
+
+    config = Jsons.clone(configWithoutDbName);
+    ((ObjectNode) config).put("database", dbName);
+  }
+
+  @Override
+  protected void tearDown(TestDestinationEnv testEnv) {}
```

Comment:
should the tear down try to remove the tables? i guess since setup does it we don't technically need to worry about it, but it is still more predictable if tests do clean themselves up. probably just putting line 174 in a helper.

Reply:
There's no good way for the tests to clean up after themselves.  Look at the functionality provided by `AirbyteDestination` and you'll see what I mean.  On our original Destination (the Postgres one), this was handled by simply recycling the container each time, so it wasn't an issue until we started working with heavier DB containers.

Reply:
fair enough. 
1. first priority is to get these tests to clean up properly so that each test runs in a hermetic environment.
2. second priority is speed. given how slow the start up is on the container it would be nice if we didn't turn it on and off every time time, but if there's no reasonable way to query the system schema for all of the tables and truncate / delete them, then i guess that's what we have to do.